### PR TITLE
Planner small fixes

### LINKED ISSUE #44490: Not found column ... in block with new analyser
```
SELECT
    sum(a.number) AS total,
    c.number AS cn,
    b.number AS bn,
    grouping(c.number) + grouping(b.number) AS l,
    rank() OVER (PARTITION BY grouping(c.number) + grouping(b.number), multiIf(grouping(c.number) = 0, b.number, NULL) ORDER BY sum(a.number) DESC) AS r
FROM numbers(10) AS a, numbers(10) AS b, numbers(10) AS c
GROUP BY
    cn,
    bn
    WITH ROLLUP
LIMIT 100
SETTINGS allow_experimental_analyzer = 1

Query id: b6102535-438d-4ed8-b14f-8adc0357ec62


0 rows in set. Elapsed: 0.002 sec.

Received exception from server (version 22.13.1):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column a.number_0 in block. There are only columns: c.number_1, b.number_2, sum(a.number_0). Stack trace:

0. ./build_docker/../src/Common/Exception.cpp:77: DB::Exception::Exception(DB::Exception::MessageMasked const&, int, bool) @ 0xd20361a in /usr/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, bool) @ 0x80b626d in /usr/bin/clickhouse
2. ./build_docker/../src/Core/Block.cpp:0: DB::Block::getByName(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool) const @ 0x11e28a2e in /usr/bin/clickhouse
3. ./build_docker/../contrib/llvm-project/libcxx/include/vector:1457: DB::Aggregator::Params::getHeader(DB::Block const&, bool, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, std::__1::vector<DB::AggregateDescription, std::__1::allocator<DB::AggregateDescription>> const&, bool) @ 0x124e3324 in /usr/bin/clickhouse
4. ./build_docker/../src/Processors/QueryPlan/RollupStep.cpp:26: DB::RollupStep::RollupStep(DB::DataStream const&, DB::Aggregator::Params, bool, bool) @ 0x140c39d1 in /usr/bin/clickhouse
5. ./build_docker/../contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: DB::Planner::buildQueryPlanIfNeeded() @ 0x12e561df in /usr/bin/clickhouse
6. ./build_docker/../src/Planner/Planner.h:38: DB::InterpreterSelectQueryAnalyzer::execute() @ 0x12e5231f in /usr/bin/clickhouse
7. ./build_docker/../src/Interpreters/executeQuery.cpp:0: DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x1318c5d0 in /usr/bin/clickhouse
8. ./build_docker/../src/Interpreters/executeQuery.cpp:1083: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x131897cd in /usr/bin/clickhouse
9. ./build_docker/../src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x13ccd00f in /usr/bin/clickhouse
10. ./build_docker/../src/Server/TCPHandler.cpp:1922: DB::TCPHandler::run() @ 0x13ce01d9 in /usr/bin/clickhouse
11. ./build_docker/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x16a30d54 in /usr/bin/clickhouse
12. ./build_docker/../contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:48: Poco::Net::TCPServerDispatcher::run() @ 0x16a3265b in /usr/bin/clickhouse
13. ./build_docker/../contrib/poco/Foundation/src/ThreadPool.cpp:213: Poco::PooledThread::run() @ 0x16bb9847 in /usr/bin/clickhouse
14. ./build_docker/../contrib/poco/Foundation/include/Poco/SharedPtr.h:277: Poco::ThreadImpl::runnableEntry(void*) @ 0x16bb727d in /usr/bin/clickhouse
15. ? @ 0x7f0c04c30609 in ?
16. clone @ 0x7f0c04b55133 in ?
. (NOT_FOUND_COLUMN_IN_BLOCK)
```

cc: @novikd @kitaisreal

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Closes #44490.

### CODE REVIEW

File: src/Planner/PlannerAggregation.cpp [hunk 1]
```diff
@@ -183,6 +185,87 @@ void resolveGroupingFunctions(QueryTreeNodePtr & query_node,
...
+                        continue;
+
+                    aggregation_keys.push_back(grouping_set_key_name);
+                    used_aggregation_keys.insert(aggregation_keys.back());
+                }
+            }
+
+            for (auto & grouping_sets_parameter : grouping_sets_parameters_list)
+            {
+                NameSet grouping_sets_used_keys;
```
Comment (novikd) [inline src/Planner/PlannerAggregation.cpp:227]: Why do we use names? Maybe it's better to do it via query tree nodes in the new analyzer?

File: src/Planner/PlannerAggregation.cpp [hunk 2]
```diff
@@ -183,6 +185,87 @@ void resolveGroupingFunctions(QueryTreeNodePtr & query_node,
     resolveGroupingFunctions(query_node, group_by_kind, aggregation_keys, grouping_sets_parameters_list, planner_context);
 }
 
+}
+
+void resolveGroupingFunctions(QueryTreeNodePtr & query_node, const PlannerContext & planner_context)
```
Comment (novikd) [inline src/Planner/PlannerAggregation.cpp:190]: Why do we resolve `GROUPING` function in the planner, not in QueryAnalysisPass? I'd recommend moving this logic there.

**Review by novikd — COMMENTED** [review]
In general LGTM, but I think logic around `GROUPING` function should be moved to the QueryAnalysisPass and rewritten using QueryTreeNodes instead of names.

File: src/Planner/PlannerAggregation.cpp (hunk 1 above)
Comment (kitaisreal) [reply src/Planner/PlannerAggregation.cpp]: Updated.

**Pull request merged**